include/device.h

```diff
@@ -85,6 +85,25 @@ extern "C" {
 	DEVICE_AND_API_INIT(dev_name, drv_name, init_fn,\
 	data, cfg_info, level, prio, NULL)
 
+/**
+ * @def DEVICE_INIT
+ *
+ * @brief Call specified macro for all device nodes corresponding to driver
+ *
+ * @details This macro will scan for all DT_INST_##index##_##driver_name
+ * nodes and call the supplied macro with the 3 parameters prefix, index, suffix
+ * where prefix is always DT_INST_, index is the instance number and suffix is
+ * always driver_name. This macro can be used to call the init macro of a driver
+ * for each device specified in the device tree.
+ *
+ * @param init_func Macro that is called for each device node. Has to accept
```

Comment:
I would name this inst_func instead

Reply:
I changed it. I am not 100% sure though. They macro is called DEVICE_INIT_ALL_INSTANCES so it assumes the passed macro is used to initialize the driver.
In theory we could rename the macro to: DEVICE_CALL_PER_INSTANCE or something like that, since it is not limited to initialization.

Reply:
The pb with init_func is that it is also a name used for existing macros. It's called init_fn so it 2 letters different, but it's confusing.

Or cal it init_macro, since your init_func should be macro.

Reply:
I change it to DEVICE_EXEC_FOR_ALL_INSTANCEs and renamed it to inst_macro.
Thanks for the input.